assets/auto-vj-uri.html

```diff
@@ -0,0 +1,141 @@
+
+               server {
+               listen       8081;
+               ...
+               location / {
+                  add_header 'Access-Control-Allow-Origin' '*' always;
+               }
+               ...
+
+               CORS header is a must so you can load videos from another hostname and/or port.
+               This example requires videos to be at "<nginx root>/html/videos/".
+            */
+            const LOCAL_URI = "http://localhost:8081/videos/"
+
+            var players = {};
+
+            function tryCreatePlayer(meta) {
+                var vLayer = document.createElement('video');
+                vLayer.muted = true;
+                vLayer.autoplay = false;
+                vLayer.defaultPlaybackRate = meta.tempo;
+                vLayer.playbackRate = meta.tempo;
+                vLayer.addEventListener('loadeddata', () => {
+                    if(vLayer.readyState == HTMLMediaElement.HAVE_ENOUGH_DATA && players[meta.filePath] && !vLayer.didStart) { // can play 100% and not removed yet
+                        document.getElementById('container').appendChild(vLayer);
+                        vLayer.play();
+                        vLayer.didStart = true;
+                    }
+                });
+                vLayer.addEventListener('ended',() => {
+                    vLayer.style.display = "none";
+                },false);
+                
+                var uReader = new XMLHttpRequest();
+                uReader.open('GET', getFilename(meta));
+                uReader.responseType = 'text'
+                uReader.onload = () => {
+                    videofilename = LOCAL_URI + uReader.responseText;
+                    const extensions = [".webm", ".mp4"];
+                    for (i = 0; i < extensions.length; i++) {
+                        videopath = videofilename + extensions[i];
+                        console.log("Searching for video file: " + videopath);
+                        if(doesFileExist(videopath)){
+                            vLayer.src = videopath;
+                            break;
+                        }       
+                    }
+                };                                                                                                                                                                                                                                                                                                                                                                                                                                           
+                uReader.send();
+                return vLayer;
+            }
+
+            // Check if URI file exist
+            function doesFileExist(urlToFile)
+            {
+                var xhr = new XMLHttpRequest();
+                xhr.open('HEAD', urlToFile, false);
+                xhr.send();
```

Comment:
I'd assume Chromium sends a HEAD before loading the video file by default for checking caches et al, so is this necessary?

Reply:
> I'd assume Chromium sends a HEAD before loading the video file by default for checking caches et al, so is this necessary?

This is used to load the first filename as '.webm' or '.mp4' if found.
I don't know and didn't test what would happen if you just change the "src" without checking if it exist, I just guessed it would not load any video at all if the last file in the iteration doesn't exist so I added this sanity check.